tpm2/seal.go

```diff
@@ -663,17 +671,49 @@ func (k *SealedKeyObject) UpdatePCRProtectionPolicy(tpm *Connection, authKey Pol
 	return updateKeyPCRProtectionPolicyCommon(tpm.TPMContext, []*SealedKeyObject{k}, ecdsaAuthKey, pcrProfile, tpm.HmacSession())
 }
 
+// RevokeOldPCRProtectionPolicies revokes old PCR protection policies associated with this sealed key. If the key
+// data was not created with a PCR policy counter, then this function does nothing. The caller must also specify
+// the private part of the authorization key that was either returned by SealKeyToTPM or SealedKeyObject.UnsealFromTPM.
+//
+// If validation of the key data fails, a InvalidKeyFileError error will be returned.
+func (k *SealedKeyObject) RevokeOldPCRProtectionPolicies(tpm *Connection, authKey PolicyAuthKey) error {
```

Comment:
there's no comment nor check that is only applied on the result of UpdatePCRProtectionPolicy, afaict this will not revoke things if that's not the case because the counter will not have been incremented in the policy data?

Maybe it just needs a better doc comment, but my point I suppose is that this does what the name implies only if k.data.dynamicPolicyData.policyCount had changed